cmd/dep/init.go

```diff
@@ -135,32 +135,33 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 		)
 	}
 
-	if len(pd.notondisk) > 0 {
-		internal.Vlogf("Solving...")
-		params := gps.SolveParameters{
-			RootDir:         root,
-			RootPackageTree: pkgT,
-			Manifest:        m,
-			Lock:            l,
-			ProjectAnalyzer: dep.Analyzer{},
-		}
-
-		if *verbose {
-			params.Trace = true
-			params.TraceLogger = log.New(os.Stderr, "", 0)
-		}
-		s, err := gps.Prepare(params, sm)
-		if err != nil {
-			return errors.Wrap(err, "prepare solver")
+	// Run solver with project versions found on disk
+	internal.Vlogf("Solving...")
+	soln, err := getSolverSolution(root, pkgT, m, l, sm)
+	if err != nil {
+		handleAllTheFailuresOfTheWorld(err)
+		return err
+	}
+	l = dep.LockFromInterface(soln)
+
+	// Pick notondisk project constraints from solution and add to manifest
+	for k, _ := range pd.notondisk {
+		for _, x := range l.Projects() {
+			if k == x.Ident().ProjectRoot {
+				m.Dependencies[k] = getProjectPropertiesFromVersion(x.Version())
+				break
+			}
 		}
+	}
 
-		soln, err := s.Solve()
-		if err != nil {
-			handleAllTheFailuresOfTheWorld(err)
-			return err
-		}
-		l = dep.LockFromInterface(soln)
+	// Run solver again with appropriate constraint solutions from previous run
```

Comment:
I don't _think_ it's necessary to solve twice. It should be sufficient to `gps.Prepare()` a new solver with the returned solution and the updated manifest, then call `solver.HashInputs()` and use that as the `Memo` value instead of what was returned from the initial solve.

More details about how this hash works, and what it means, are [here](https://github.com/sdboyer/gps/wiki/gps-for-Implementors#solvehashinputs).